dashboard/app/models/school.rb

```diff
@@ -0,0 +1,60 @@
+# == Schema Information
+#
+# Table name: schools
+#
+#  id                 :integer
+#  school_district_id :integer
+#  name               :string(255)
+#  school_type        :string(255)
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# Indexes
+#
+#  index_schools_on_school_district_id  (school_district_id)
+#
+
+class School < ActiveRecord::Base
+  include Seeded
+
+  # The listing of all US schools comes from http://nces.ed.gov/ccd/pubagency.asp
+  # and is then exported into a tab-separated file.
+  # The data format is described at http://nces.ed.gov/ccd/pdf/2015150_sc132a_Documentation_052716.pdf
+  CSV_HEADERS = {
+    :id => 'NCESSCH',
+    :school_district_id => 'LEAID',
+    :name => 'SCHNAM',
+    :charter_status => 'CHARTR',
+  }
+
+  # Use the zero byte as the quote character to allow importing double quotes
+  #   via http://stackoverflow.com/questions/8073920/importing-csv-quoting-error-is-driving-me-nuts
+  CSV_IMPORT_OPTIONS = { col_sep: "\t", headers: true, quote_char: "\x00" }
+
+  def self.find_or_create_all_from_tsv!(filename)
+    created = []
+    CSV.read(filename, CSV_IMPORT_OPTIONS).each do |row|
+      created << self.first_or_create_from_tsv_row!(row)
+    end
+    created
+  end
+
+  SCHOOL_TYPE = {
+    public: 'public',
+    charter: 'charter'
+  }
+
+  def self.school_type(charter_status)
+    charter_status == '1' ? SCHOOL_TYPE[:charter] : SCHOOL_TYPE[:public]
+  end
```

Comment:
It looks like only instance methods (`def foo`) can be made private whereas these are class methods (`def self.bar`). This is what I get in irb:
```
irb(main):038:0> class MyClass
irb(main):039:1>   private
irb(main):040:1> 
irb(main):041:1*   def foo
irb(main):042:2>     puts "foo"
irb(main):043:2>   end
irb(main):044:1> 
irb(main):045:1*   def self.bar
irb(main):046:2>     puts "bar"
irb(main):047:2>   end
irb(main):048:1> end
=> :bar
irb(main):049:0> MyClass.new.foo
NoMethodError: private method `foo' called for #<MyClass:0x007fe31b1d3048>
	from (irb):49
	from /Users/dsb/.rbenv/versions/2.2.3/bin/irb:11:in `<main>'
irb(main):050:0> MyClass.bar
bar
=> nil
```

However, `private_class_method` is a thing in ruby that we apparently rarely use but seems appropriate here, so I've changed it to use that.